pkg/controller/podgroup_test.go

```diff
@@ -183,12 +183,98 @@ func Test_Run(t *testing.T) {
 
 }
 
-func makePods(podNames []string, pgName string, phase v1.PodPhase) []*v1.Pod {
+func TestFillGroupStatusOccupied(t *testing.T) {
+	ctx := context.TODO()
+	cases := []struct {
+		name                 string
+		pgName               string
+		minMember            int32
+		podNames             []string
+		podPhase             v1.PodPhase
+		podOwnerReference    []metav1.OwnerReference
+		groupPhase           v1alpha1.PodGroupPhase
+		desiredGroupOccupied string
+	}{
+		{
+			name:      "fill the desired Occupied",
+			pgName:    "pg",
+			minMember: 2,
+			podNames:  []string{"pod1", "pod2"},
+			podPhase:  v1.PodPending,
+			podOwnerReference: []metav1.OwnerReference{
+				{
+					Name: "new-occupied",
+				},
+			},
+			groupPhase:           v1alpha1.PodGroupPending,
+			desiredGroupOccupied: "default/new-occupied",
+		},
+		{
+			name:      "fill the desired Occupied, with multi ownerReference",
+			pgName:    "pg",
+			minMember: 2,
+			podNames:  []string{"pod1", "pod2"},
+			podPhase:  v1.PodPending,
+			podOwnerReference: []metav1.OwnerReference{
+				{
+					Name: "new-occupied-1",
+				},
+				{
+					Name: "new-occupied-2",
+				},
+			},
+			groupPhase:           v1alpha1.PodGroupPending,
+			desiredGroupOccupied: "default/new-occupied-1,default/new-occupied-2",
+		},
+	}
+	for _, c := range cases {
+		t.Run(c.name, func(t *testing.T) {
+			var kubeClient *fake.Clientset
+			if len(c.podNames) == 0 {
+				kubeClient = fake.NewSimpleClientset()
+			} else {
+				ps := makePods(c.podNames, c.pgName, c.podPhase, c.podOwnerReference)
+				kubeClient = fake.NewSimpleClientset(ps[0], ps[1])
+			}
+			pg := makePG(c.pgName, c.minMember, c.groupPhase, nil)
+			pgClient := pgfake.NewSimpleClientset(pg)
+
+			informerFactory := informers.NewSharedInformerFactory(kubeClient, controller.NoResyncPeriodFunc())
+			pgInformerFactory := schedinformer.NewSharedInformerFactory(pgClient, controller.NoResyncPeriodFunc())
+			podInformer := informerFactory.Core().V1().Pods()
+			pgInformer := pgInformerFactory.Scheduling().V1alpha1().PodGroups()
+			ctrl := NewPodGroupController(kubeClient, pgInformer, podInformer, pgClient)
```

Comment:
This part of the code has a lot of duplication with another test.
It's better to define a separate function for `setup env for test`.

Reply:
> It's better to define a separate function for setup env for test.

We can leave it to the kubebuilder migration. (@helayoty is working on using kubebuilder and envtest to migrate existing controller-related code).

Reply:
> We can leave it to the kubebuilder migration. 
Sure

Reply:
tagging the kubebuilder issue here so I remember to handle that #327